hack/jenkins/e2e-runner.sh

```diff
@@ -160,7 +160,15 @@ if [[ "${gcp_list_resources}" == "true" ]]; then
   ${gcp_list_resources_script} > "${gcp_resources_before}"
 fi
 if [[ "${E2E_UP,,}" == "true" ]]; then
-    go run ./hack/e2e.go ${E2E_OPT:-} -v --up
+    # We want to try to gather logs even if kube-up fails, so collect the
+    # result here and fail after dumping logs if it's nonzero.
+    go run ./hack/e2e.go ${E2E_OPT:-} -v --up && up_result="$?" || up_result="$?"
+    if [[ -x "cluster/log-dump.sh"  ]]; then
+        ./cluster/log-dump.sh "${ARTIFACTS}"
```

Comment:
we're going to end up with a lot of duplicated logs, before and after e2e tests run, but it's currently not very clear which logs are which (unless you look at timestamps). should we maybe only collect logs here if we fail to come up? or use a subdir or something?